src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -282,28 +283,31 @@ public void runAppCfgCommand(List<String> args) throws ProcessRunnerException {
    * Returns the version of the Cloud SDK installation. Version is determined by reading the VERSION
    * file located in the Cloud SDK directory.
    *
-   * @throws CloudSdkOutOfDateException if the Cloud SDK is too out of date to determine its version
-   * @throws IOException if the VERSION file could not be read. This might mean the Cloud SDK
-   *                     installation is corrupted.
+   * @throws CloudSdkVersionFileNotFoundException if the VERSION file is not present
+   * @throws RuntimeException if there was an error reading the file
+   * @throws IllegalStateException if the file content could not be parsed
    */
-  public CloudSdkVersion getVersion() throws IOException {
-    Path versionFile = getSdkPath().resolve(VERSION_FILE);
+  public CloudSdkVersion getVersion() {
+    Path versionFile = getSdkPath().resolve(VERSION_FILE_NAME);
 
     if (!Files.isRegularFile(versionFile)) {
-      throw new CloudSdkOutOfDateException(MINIMUM_VERSION);
+      throw new CloudSdkVersionFileNotFoundException("Cloud SDK version file not found at "
```

Comment:
I may be missing something but I don't see much point to this exception if its just thrown and caught internally in this class.